src/cargo/util/toml/targets.rs

```diff
@@ -402,15 +559,60 @@ fn clean_benches(
             errors,
             &mut legacy_bench_path,
             "autobenches",
-        )?
+        )?;
+
+        let mut targets: Vec<_> = targets.into_iter()
+            .map(|(path, target)| (path, None, target))
+            .collect();
+
+        targets.extend(
+            toml_targets.iter().flat_map(|ps| {
+                ps.iter()
+                    .flat_map(|(cfg, platform)| {
+                        platform.bench
+                            .iter()
+                            .flat_map(|benches| {
+                                benches.iter()
+                                    .cloned()
+                                    .filter_map(|bench| {
```

Comment:
This is a particularly gnarly super-nested loop duplicated (it looks like) in a few places, could these be tightened up?